code/parachain/frame/assets-registry/src/tests.rs

```diff
@@ -64,27 +70,33 @@ fn register_asset() {
 		let location = <Runtime as crate::Config>::ForeignAssetId::decode(
 			&mut &XcmAssetLocation::RELAY_NATIVE.encode()[..],
 		)
-		.unwrap();
-		let ratio = rational!(42 / 123);
-		let decimals = 3;
+		.expect("Location bytes translate to foreign ID bytes");
+		let asset_info = AssetInfo {
+			name: None,
+			symbol: None,
+			decimals: 4,
```

Comment:
there are assets with not know decimals yet, but that does not means they would not have well known decimals in future. So value can be set and never change Some(0,1, 12, 18, ..)  or None(so it can later be set after metadata provider (Acala supports such tokens and Statemine)

Reply:
So, we should ensure the interface supports updating the decimals of an asset and initially setting the decimals to `None`?

Reply:
yeah. we should generally be able to handle assets without metadata (unknown assets) and accept all tokens (having ED for these is also should be solved at some point)